projects/biogears/libBiogears/src/engine/Systems/Drugs.cpp

```diff
@@ -384,6 +385,104 @@ void Drugs::AdministerSubstanceInfusion()
 
 //-------------------------------------------------------------------------------------------------
 /// \brief
+/// Administer drugs via nasal route
+///
+/// \details
+/// This function administers drug into the nose as substance that can interact with three different
+/// sections: anterior, posterior, and gastrointestinal. Drug is initially deposited in the anterior
+/// and posterior sections and goes through rate constant-determined translocation into either uptake
+/// or excretion through the GI section. This administration path is generalized and assumes that the
+/// drug has a carrier, however, adjustment of drug release constants allows for administration with
+/// non-carrier drugs
+//-------------------------------------------------------------------------------------------------
+void Drugs::AdministerSubstanceNasal()
+{
+  SESubstance* naloxone = m_data.GetSubstances().GetSubstance("Naloxone");
+
+  const std::map<const SESubstance*, SESubstanceNasalDose*>& nasalDoses = m_data.GetActions().GetPatientActions().GetSubstanceNasalDoses();
+  if (nasalDoses.empty())
+    return;
+
+  double nasaltime = nasalDoses.at(naloxone)->GetElapsedTime().GetValue(TimeUnit::s);
+
+  double nasaldose = nasalDoses.at(naloxone)->GetDose().GetValue(MassUnit::mg);
+
+  //Rate constants in 1/s
+  const double nasalk1 = 0.00001736111;  //translocation rate constant of unreleased substance from the anterior to the posterior section
```

Comment:
everything looks great Rishi but can you tag the paper you used to get these values real quick. WE will then work to merge in

Reply:
[Gonda, I., Gipps, E. Model of Disposition of Drugs Administered into the Human Nasal Cavity. Pharm Res 7, 69–75 (1990).](https://pubmed.ncbi.nlm.nih.gov/2300539/)